modules/public/public.go

```diff
@@ -99,6 +106,19 @@ func (opts *Options) handle(ctx *macaron.Context, log *log.Logger, opt *Options)
 
 	f, err := opt.FileSystem.Open(file)
 	if err != nil {
+		// 404 requests to any known entries in `public`
+		if path.Base(opts.Directory) == "public" {
```

Comment:
This condition seems to be needed because this handler is hit twice on every request, once with `opts.Directory` as `""` and once with a actual path.